src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -164,12 +168,18 @@ public void serializeToView(@NotNull ByteBuf buf)
     {
         super.serializeToView(buf);
 
+        updateAvailableResources();
         buf.writeInt(neededResources.size());
-
-        for (@NotNull final Map.Entry<Block, Integer> entry : neededResources.entrySet())
+        for (@NotNull final Map.Entry<String, BuildingBuilderResource> entry : neededResources.entrySet())
         {
-            ByteBufUtils.writeUTF8String(buf, entry.getKey().getLocalizedName());
-            buf.writeInt(entry.getValue());
+            final BuildingBuilderResource resource = neededResources.get(entry.getKey());
+            //ByteBufUtils.writeItemStack() is Buggy, serialize itemId and damage separately;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)